weblab/repocache/models.py

```diff
@@ -67,6 +67,22 @@ def get_version(self, sha):
         except ObjectDoesNotExist:
             raise RepoCacheMiss("Entity version not found")
 
+    def get_name_for_version(self, sha):
+        """Get a human-friendly display name for the given version
+
+        :param sha: version sha
+        :return: first cached tag for this version, if any, or sha if not
+        """
+        try:
+            version = self.get_version(sha)
+        except ObjectDoesNotExist:
+            raise RepoCacheMiss("Entity version not found")
```

Comment:
`self.get_version` already raises `RepoCacheMiss` so you don't need the try-except here, just call the method.

(Hence why [coverage](https://codecov.io/gh/ModellingWebLab/WebLab/pull/240/changes#D1-70) shows an uncovered line.)

Reply:
thanks all very helpful - appreciate it.

Reply:
Its interesting.. get_commit() supportted getting a commit by tag name. I suppose I am going to have to support that in get_version() too.

Reply:
Yes - it's basically the converse of the get_nice_name, translating that name back into an actual commit.

Reply:
The `string_if_invalid` works very well!! - 
![image](https://user-images.githubusercontent.com/3098326/73171918-bf601180-40f9-11ea-9209-8c924b8152d7.png)


![image](https://user-images.githubusercontent.com/3098326/73171895-b2432280-40f9-11ea-8305-351637ce8675.png)